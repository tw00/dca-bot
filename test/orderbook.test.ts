/* eslint-disable @typescript-eslint/no-empty-function */
import Simulation from "../lib/simulation";
import Order from "../lib/order";

describe("orderbook", () => {
  it("should work", async () => {
    const sim = new Simulation({ symbol: "BTC", fee: 0 });

    const testorder1 = Order.Buy(0.01, "BTC").atStopPrice(56205);
    const testorder2 = Order.Sell(0.05, "BTC").atStopPrice(56202);

    sim.exchange.addOrder(testorder1);
    sim.exchange.addOrder(testorder2);

    const backupConsoleLog = console.log;
    console.log = () => {};
    console.warn = jest.fn(() => {});

    await sim.init();
    await sim.run();

    expect(console.warn).toHaveBeenCalled();

    const res = [];
    console.log = (...args) =>
      res.push(args.map((x) => JSON.stringify(x, null, 2)).join(" "));
    sim.exchange.portfolio.print();
    console.log = backupConsoleLog;
    expect(res).toMatchSnapshot();
  });
});
