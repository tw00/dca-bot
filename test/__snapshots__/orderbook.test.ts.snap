// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderbook should work 1`] = `
Array [
  "\\"========================================\\"",
  "\\"Transactions:\\" [
  {
    \\"amount\\": 1000,
    \\"symbol\\": \\"USD\\",
    \\"price\\": 1
  },
  {
    \\"amount\\": 0.01,
    \\"symbol\\": \\"BTC\\",
    \\"price\\": 56203.88
  },
  {
    \\"amount\\": -563.443897,
    \\"symbol\\": \\"USD\\",
    \\"price\\": 56203.88
  },
  {
    \\"amount\\": -0.01,
    \\"symbol\\": \\"BTC\\",
    \\"price\\": 56202.01
  },
  {
    \\"amount\\": 562.0201000000001,
    \\"symbol\\": \\"USD\\",
    \\"price\\": 56202.01
  }
]",
  "\\"Positions:\\" {
  \\"USD\\": 998.5762030000001,
  \\"BTC\\": 0
}",
  "\\"========================================\\"",
]
`;
